x-pack/plugins/cloud/server/plugin.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { get } from 'lodash';
+import { first } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+import { UsageCollectionSetup } from 'src/plugins/usage_collection/server';
+import { CoreSetup, Logger, Plugin, PluginInitializerContext } from 'src/core/server';
+import { CloudConfigType } from './config';
+import { registerCloudUsageCollector } from './collectors';
+import { getIsCloudEnabled } from '../common/is_cloud_enabled';
+
+interface PluginsSetup {
+  usageCollection?: UsageCollectionSetup;
+}
+
+export interface CloudSetup {
+  cloudId?: string;
+  isCloudEnabled: boolean;
+  apm: {
+    url?: string;
+    secret_token?: string;
```

Comment:
Probably getting a type error here, but this is snakecase here and camelcase below